src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -322,6 +322,22 @@ class ErgoWalletActor(settings: ErgoSettings,
       walletVars = walletVars.resetProver()
       secretStorageOpt.foreach(_.lock())
 
+    case RescanWallet =>
+      // We do wallet rescan by closing the wallet's database, deleting it from the disk,
+      // then reopening it and sending a rescan signal.
+      val rescanResult = Try {
+        val registryFolder = WalletRegistry.registryFolder(settings)
+        log.info(s"Rescanning the wallet, the registry is in $registryFolder")
+        registry.close()
+        FileUtils.deleteRecursive(registryFolder)
+        registry = WalletRegistry.apply(settings)
+        self ! ScanInThePast(walletHeight()) // walletHeight() corresponds to empty wallet state now
```

Comment:
This is async sending message. When Try is finished, the rescan is not completed yet. Is it by design?

Reply:
@aslesarenko yes, rescan takes long time and must be performed asynchronously